runway/core/providers/docker/models/images.py

```diff
@@ -0,0 +1,129 @@
+                unsuccessful builds
+            dockerfile (str): path within the build context to the Dockerfile
+            buildargs (dict): A dictionary of build arguments
+            container_limits (dict): A dictionary of limits applied to each
+                container created by the build process. Valid keys:
+                - memory (int): set memory limit for build
+                - memswap (int): Total memory (memory + swap), -1 to disable
+                    swap
+                - cpushares (int): CPU shares (relative weight)
+                - cpusetcpus (str): CPUs in which to allow execution, e.g.,
+                    ``"0-3"``, ``"0,1"``
+            shmsize (int): Size of `/dev/shm` in bytes. The size must be
+                greater than 0. If omitted the system uses 64MB
+            labels (dict): A dictionary of labels to set on the image
+            cache_from (list): A list of images used for build cache
+                resolution
+            target (str): Name of the build-stage to build in a multi-stage
+                Dockerfile
+            network_mode (str): networking mode for the run commands during
+                build
+            squash (bool): Squash the resulting images layers into a
+                single layer.
+            extra_hosts (dict): Extra hosts to add to /etc/hosts in building
+                containers, as a mapping of hostname to IP address.
+            platform (str): Platform in the format ``os[/arch[/variant]]``.
+            isolation (str): Isolation technology used during build.
+                Default: `None`.
+            use_config_proxy (bool): If ``True``, and if the docker client
+                configuration file (``~/.docker/config.json`` by default)
+                contains a proxy configuration, the corresponding environment
+                variables will be set in the container being built.
+
+        Raises:
+            :py:class:`docker.errors.BuildError`
+                If there is an error during the build.
+            :py:class:`docker.errors.APIError`
+                If the server returns any other error.
+            ``TypeError``
+                If neither ``path`` nor ``fileobj`` is specified.
+
+        """
+        resp = self.client.api.build(**kwargs)
+        if isinstance(resp, six.string_types):
+            return self.get(resp)
+        last_event = None
+        image_id = None
+        result_stream, internal_stream = itertools.tee(json_stream(resp))
+        # added real-time logging
+        for chunk in internal_stream:
+            if "error" in chunk:
+                LOGGER.error(chunk["error"].strip())
+                raise BuildError(chunk["error"], result_stream)
+            if "stream" in chunk:
+                stream_msg = chunk["stream"]
+                if re.search(r"^Step \d*\/\d* : \w", stream_msg):
+                    LOGGER.info(stream_msg.strip())
```

Comment:
Can we PR in these logging options upstream, instead of vendoring it here? I get how much more informative it is, but it'd  be really nice to just be able to call the library methods directly.

Reply:
I agree that it would be nice but I highly doubt it would be accepted upstream. They don't log anything outside of internal debug messages (nothing from the Docker service itself). _Most_ of the SDK returns a generator to iterate over for real-time logs. This one does not since it needs to inspect the logs to get the image ID.

Reply:
> They don't log anything outside of internal debug messages

This was determined by searching their codebase for uses of `logging.getLogger` (used in 6 places). They assign their logger to the variable `log`. Searching for `log.error`, `log.warning`, and `log.error` returned no results. Searching for `log.debug` returned 5 results, none of which log anything returned from the Docker API. Only internal messages.

@troyready - if you want me to try submitting a PR upstream I can but given the above it probably won't go anywhere. Just trying to prevent this from being a blocker here or in the docker-py repo. I can remove the logging completely if it's an issue. It's just not as "user friendly" especially for long running builds with a lot of steps (or those of us that panic a little when the terminal stops providing feedback for more than 3 seconds) as the terminal will just sit there and not be able to provide feedback until the `docker image build` action is complete.

Reply:
Sorry for the delayed response.

Please do remove the logging. I don't think there's a perfect answer here, but I'd rather future us have a bit more distance from the internals of docker-py.